crates/trigger/src/runtime_config.rs

```diff
@@ -92,6 +96,29 @@ impl RuntimeConfig {
             .unwrap_or_else(|| KeyValueStoreOpts::default_store_opts(self))
     }
 
+    /// Return an iterator of named configured [`SqliteDatabase`]s.
+    pub fn sqlite_databases(
+        &self,
+    ) -> Result<impl IntoIterator<Item = (String, sqlite::SqliteDatabase)>> {
+        let mut databases = HashMap::new();
+        // Insert explicitly-configured databases
+        for opts in self.opts_layers() {
+            for (name, database) in &opts.sqlite_databases {
+                if !databases.contains_key(name) {
+                    let store = database.build(name, opts)?;
+                    databases.insert(name.to_owned(), store);
+                }
+            }
+        }
+        // Upsert default store
+        if !databases.contains_key("default") {
```

Comment:
(Optional): could use `HashMap::entry` and https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html#method.or_insert_with here and above instead of the `contains_key`/`insert` combo.

Reply:
Entry requires an owned key every time so we would do an extra allocation. I find in this particular case the non-entry based code is easier to read too. I think I'd prefer to keep it as it is, but let me know if you disagree!